src/typechecker/FStar.TypeChecker.TcTerm.fst

```diff
@@ -3983,6 +3986,10 @@ and check_top_level_let_rec env top =
                                         lb.lbpos)
                               ecs
                               lbs
+                in
+                (* discharge generalization uvars *)
+                let g_lbs = Rel.resolve_generalization_implicits env g_lbs in
```

Comment:
Is it clear here that this is called only in phase1? Since this function explicitly does not typecheck solutions to generalized uvars.

I guess env.generalize is only set in phase1 or lax?

Reply:
> I guess env.generalize is only set in phase1 or lax?

Yes, exactly.